### ISSUE TITLE
.NET Core 1.0.3

### ISSUE THREAD

**Issue by Thealexbarney** [appveyor/ci#1262]
I've noticed that the .NET Core runtime and SDKs for 1.0.0, 1.0.1, and 1.1.0 are on the build images, but the most recent LTS release, 1.0.3, isn't. Are there any plans to add this to the build images?

**Comment by FeodorFitsner** [comment]
Are you experiencing an issue while building 1.0.3 apps on AppVeyor? Can you give us "hello world" example of such app?

**Comment by Thealexbarney** [comment]
[Here's](https://github.com/Thealexbarney/dotnet-1.0.3-appveyor) an example of a test project created with `dotnet new`, and its [AppVeyor ](https://ci.appveyor.com/project/Thealexbarney/dotnet-1-0-3-appveyor)page.

As seen in the build history, it's unable to test the project because it doesn't have the latest framework. The workaround is to download the SDK and runtime before each build.

**Comment by FeodorFitsner** [comment]
Can you help me to understand please why this app is building on current image: https://github.com/appveyor-tests/dotnet-core/tree/master/WebAppDotNetCore103/src/WebAppDotNetCore103
Results: https://ci.appveyor.com/project/appveyor-tests/dotnet-core

**Comment by Thealexbarney** [comment]
It'll build with the preview 2.1 1.1.0 SDK, but would be unable to run the project or run tests because the 1.0.3 runtime is missing.

**Comment by Thealexbarney** [comment]
And the latest [LTS SDK release](https://www.microsoft.com/net/download/core), 1.0.0-preview2-003156, isn't available, so it has to be manually downloaded if you want to use the LTS branch.

**Comment by FeodorFitsner** [comment]
But you can't have both 1.0 and 1.1 SDKs installed, right?

**Comment by Thealexbarney** [comment]
Each SDK and runtime is stored separately from the others. The default directories are "%PROGRAMFILES%\dotnet\sdk" and "%PROGRAMFILES%\shared\Microsoft.NETCore.App" respectively.

The current vs2015 image has SDKs "1.0.0-preview2-003121", "1.0.0-preview2-003131", and "1.0.0-preview2-1-003177". It has runtimes 1.0.0, 1.0.1, and 1.1.0. The user can choose which SDK to use at build time.

I think most of the releases can be found [here](https://github.com/dotnet/core/blob/master/release-notes/download-archive.md)

**Comment by FeodorFitsner** [comment]
Cool, good to know, thanks.

So, how would you choose specific SDK while building the app? I mean after installing SDK we usually do `dotnet new`, etc. to "warm up" and pre-cache .NET Core libs. How would we do that for each of the SDK?

**Comment by britzkopf** [comment]
Does [this](https://docs.microsoft.com/en-us/dotnet/articles/core/tools/index#driver) answer that question?

**Comment by Thealexbarney** [comment]
Currently you would specify which SDK version to use in a global.json file in your working directory. There's been talk about specifying it another way, but I don't know where that's gone recently.

I just do this to create the file:
```
function CreateBuildGlobalJson([string] $Path, [string]$Version)
{
    $json = "{`"projects`":[],`"sdk`":{`"version`":`"$Version`"}}"
    Out-File -FilePath $Path -Encoding utf8 -InputObject $json
}
```

**Comment by FeodorFitsner** [comment]
OK, got it.

So what SDKs do we have to install to build workers? All of them? For example of [this page](https://github.com/dotnet/core/blob/master/release-notes/download-archive.md) there is `SDK Preview 4 build 4233` coming with 1.0.1 (not newer 1.0.3?). Is that Preview 4 newer than `Preview 2.1 build 3177` coming with 1.1 runtime? Does it work with 1.1 runtime?

Thanks!

**Comment by FeodorFitsner** [comment]
@Thealexbarney based on your test app I did the following test checking all installed .NET Core runtimes/SDKs: https://github.com/appveyor-tests/dotnet-core

What do you think?

**Comment by FeodorFitsner** [comment]
Btw, found good article explaining .NET Core runtimes/SDKs: http://blog.tpcware.com/2016/12/multiple-versions-of-net-core-runtimes-and-sdk-tools-sxs-survive-guide/

**Comment by Thealexbarney** [comment]
It's important to note that the SDK and the runtime aren't dependent on specific versions of each other. You can pair any SDK with any runtime. The SDK is only used to build projects, and the runtime runs them.

All 1.0.x versions can be used as drop-in replacements for lower 1.0.x versions, and the same goes for 1.1.x versions.

At the very least, it would be best to have the latest SDK and runtime from each branch (LTS and Current). That is runtime 1.0.3 and SDK 1.0.0-preview2-003156 for LTS, and runtime 1.1.0 and SDK 1.0.0-preview2-1-003177 for current.

For best convenience for users, it might be good to have older SDK and runtime versions as well.

**Comment by Thealexbarney** [comment]
BTW, 1.0.0-preview2-003133 was released as an update to 1.0.0-preview2-003131. The original 1.0.0-preview2-003131 isn't listed in the download archive
